src/partition.js

```diff
@@ -1,30 +1,29 @@
-var _curry2 = require('./internal/_curry2');
-var _reduce = require('./internal/_reduce');
+var filter = require('./filter');
+var juxt = require('./juxt');
+var reject = require('./reject');
 
 
 /**
- * Takes a predicate and a list and returns the pair of lists of elements which
- * do and do not satisfy the predicate, respectively.
+ * Takes a predicate and a list or other "filterable" object and returns the
+ * pair of filterable objects of the same type of elements which do and do not
+ * satisfy, the predicate, respectively.
  *
  * @func
  * @memberOf R
  * @since v0.1.4
  * @category List
- * @sig (a -> Boolean) -> [a] -> [[a],[a]]
- * @param {Function} pred A predicate to determine which array the element belongs to.
- * @param {Array} list The array to partition.
- * @return {Array} A nested array, containing first an array of elements that satisfied the predicate,
- *         and second an array of elements that did not satisfy.
+ * @sig Filterable f => (a -> Boolean) -> f -> [f,f]
```

Comment:
We certainly want `f a`, as Scott suggested. It may be helpful to work backwards from a concrete type:

```haskell
(Number -> Boolean) -> Array Number -> Pair (Array Number) (Array Number)
```

Replace `Number` with `a`:

```haskell
(a -> Boolean) -> Array a -> Pair (Array a) (Array a)
```

Replace `Array` with `Filterable f => f`:

```haskell
Filterable f => (a -> Boolean) -> f a -> Pair (f a) (f a)
```

The remaining question is how to represent a pair. I favour `Pair a b` as it's unambiguous, whereas `(a, b)` collides with our representation of binary functions and `[a, b]` reuses our array notation in a conflicting manner. I'm okay with `[f a, f a]`, though.